src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,211 @@
+package duke;
+
+import java.io.IOException;
+import java.util.Scanner;
+
+/**
+ * A class that handles the user input commands.
+ */
+public class Parser {
```

Comment:
Would it be better for the parser to solely handle parsing of the input? Perhaps a separate Command class can handle the execution of the commands and the UI class can handle displaying of messages.